src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1371 @@
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = H(y^2, (x + y)^3 * (x^2 + x * y + y^2))
+            sage: A.reduce()
+            Arrangement (y, x^3 + 2*x^2*y + 2*x*y^2 + y^3) in Affine Space
+            of dimension 2 over Rational Field
+            sage: C = H(x*y, x*(y + 1))
+            sage: C.reduce()
+            Some curves have common components
+            sage: C.reduce(clean=True)
+            Arrangement (x*y, y + 1) in Affine Space of dimension 2
+            over Rational Field
+            sage: C = H(x*y, x)
+            sage: C.reduce(clean=True)
+            Arrangement (x*y) in Affine Space of dimension 2 over Rational Field
+        """
+        P = self.parent()
+        R = self.coordinate_ring()
+        L = [self._curves[0].defining_polynomial().radical()]
+        for c in self._curves[1:]:
+            g = c.defining_polynomial().radical()
+            for f in L:
+                d = g.gcd(f)
+                if d.degree() > 0 and not clean:
+                    print("Some curves have common components")
+                    return None
+                g //= d
+            if g.degree() > 0:
+                L.append(g)
+        return P(*L)
+
+
+class AffinePlaneCurveArrangementElement(PlaneCurveArrangementElement):
+    """
+    An ordered affine plane curve arrangement.
+    """
+    def __init__(self, parent, curves, check=True):
+        """
+        Construct an ordered affine plane curve arrangement.
+
+        INPUT:
+
+        - ``parent`` -- the parent :class:`AffinePlaneCurveArrangements`
+
+        - ``curves`` -- a tuple of curves
+
+        EXAMPLES::
+
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: elt = H(x, y); elt
+            Arrangement (x, y) in Affine Space of dimension 2 over Rational Field
+            sage: TestSuite(elt).run()
+        """
+        Element.__init__(self, parent)
+        self._curves = tuple(curves)
+        if check:
+            if not all(isinstance(h, AffinePlaneCurve) for h in curves):
+                raise ValueError("not all elements are curves")
+            if not all(h.ambient_space() is self.parent().ambient_space()
+                       for h in curves):
+                raise ValueError("not all curves are in the same ambient space")
+        self._braid_monodromy_non_vertical = None
+        self._braid_monodromy_vertical = None
+        self._strands_nonvertical = None
+        self._strands_vertical = None
+        self._fundamental_group_nonsimpl_nonvertical = None
+        self._fundamental_group_nonsimpl_vertical = None
+        self._fundamental_group_simpl_nonvertical = None
+        self._fundamental_group_simpl_vertical = None
+        self._meridians_nonsimpl_nonvertical = None
+        self._meridians_nonsimpl_vertical = None
+        self._meridians_simpl_nonvertical = None
+        self._meridians_simpl_vertical = None
```

Comment:
For these, I am thinking you should have a hidden `@cached_method` that returns both the group and meridians based on the input. Then the public facing methods just return the corresponding object of the pair.

Reply:
Hopefully we may have it after #37128